### PR TITLE
Fix bug when question is renamed to a subdirectory of itself

### PR DESCRIPTION
**Pull request by nwalters512**
See inline descriptions of the problem and the fix.

### CODE REVIEW

File: lib/chunks.js [hunk 1]
```diff
@@ -925,6 +925,27 @@ const ensureChunk = async (courseId, chunk) => {
...
+  // exists and is a symlink. If so, we'll remove it. This should always be
+  // safe because we should never have nested symlinks.
+  const pathSegments = relativeTargetPath.split(path.sep);
+  for (let i = 1; i < pathSegments.length; i++) {
+    const parentPath = path.join(courseChunksDirs.course, ...pathSegments.slice(0, i));
+    try {
+      const stat = await fs.lstat(parentPath);
+      if (stat.isSymbolicLink()) {
+        await fs.remove(parentPath);
+      }
```
Comment (mwest1066) [inline lib/chunks.js:943]: For an added check we could do something like:
```suggestion
      } else if (!stat.isDirectory()) {
        throw new Error(`${parentPath} exists but is not a directory`, stat);
      }
```

File: tests/chunks.test.js [hunk 2]
```diff
@@ -311,7 +313,60 @@ describe('chunks', () => {
...
     });
 
-    it('deletes chunks that are no longer needed', async () => {
+    it('handles question nesting after a rename', async () => {
+      // Scenario: there's a question named `foo/bar` (that is,
+      // `foo/bar/info.json` exists). We load the chunk for that
+      // question. We then move that `info.json` file to
+      // `foo/bar/baz/info.json`. We then try to load that chunk again. In the
+      // past, the new chunk would be written to an invalid location. This test
+      // ensures that it's written correctly.
```
Comment (mwest1066) [inline tests/chunks.test.js:322]: Such a sneaky scenario!

**Review by mwest1066 — APPROVED** [review]

File: lib/chunks.js (hunk 1 above)
Comment (nwalters512) [reply lib/chunks.js:943]: Ah, interesting. That might actually be a scenario in which we should do something useful? Consider the following course, where `bar` is an extensionless file:

```
├── questions
│   ├── foo
│   │   ├── info.json
│   │   ├── bar
```

At some point that's changed to the following:

```
├── questions
│   ├── foo
│   │   ├── bar
│   │   │   ├── info.json
```

I'm pretty sure our code won't handle this case; maybe it should? This is admittedly a pretty contrived situation.

Comment (nwalters512) [reply lib/chunks.js:943]: Actually it turns out that this new code handles this just fine! I'll accept your suggestion then.

**Pull request merged**